breathecode/events/views.py

```diff
@@ -725,6 +725,58 @@ def get(self, request, event_type_slug, academy_id=None):
         serializer = EventTypeVisibilitySettingSerializer(items, many=True)
         return handler.response(serializer.data)
 
+    @capable_of('crud_event_type')
+    def post(self, request, event_type_slug, academy_id=None):
+        handler = self.extensions(request)
+
+        event_type = EventType.objects.filter(slug=event_type_slug).first()
+        if not event_type:
+            raise ValidationException('Event type not found', slug='event-type-not-found')
+
+        academy = Academy.objects.filter(id=request.data['academy']).first()
+        if not academy:
+            raise ValidationException('Academy not found', slug='academy-not-found')
+
+        syllabus = None
+        if 'syllabus' in request.data:
+            syllabus = Syllabus.objects.filter(id=request.data['syllabus']).first()
+
+        cohort = None
+        if 'cohort' in request.data:
+            cohort = Cohort.objects.filter(id=request.data['cohort']).first()
+
+        visibility_setting, created = EventTypeVisibilitySetting.objects.get_or_create(syllabus=syllabus,
```

Comment:
Then, you got the instance of visibility setting, so you need add this to the many to many to `EventType` if it ain't already in it

Reply:
```py
if visibility_setting not in event_type.visibility_settings:
    event_type.visibility_settings.add(visibility_setting)
```